pkg/kubelet/status/status_manager.go

```diff
@@ -462,7 +462,7 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 	}
 
 	// We failed to update status, wait for periodic sync to retry.
-	glog.Warningf("Failed to update status for pod %q: %v", format.Pod(pod), err)
+	glog.Warningf("Failed to update status for pod %q: %v", format.PodDesc(status.podName, status.podNamespace, uid), err)
```

Comment:
The fix is good but I think we can clean this function up a little while we are at it.  Can you copy this glog line into a `if err != nil` above line 430 and return immediately so that there is no potential use of `pod` later in the function, which was the source of this issue.  Then we can remove the `if err == nil` block on line 430 and reduce indentation.  Then you can leave the message on 465 as is, since we know `pod != nil` now.